### PR TITLE
Allow minority/majority producers permission to be removed

### PR DESCRIPTION
**Pull request by heifner**
- Nothing prevents `eosio.prods` `prod.major` or `prod.minor` permissions from being deleted.
  - `cleos set account permission eosio.prods prod.minor NULL`
  - `cleos set account permission eosio.prods prod.major NULL`
- Verify permissions exist before attempting to update them in `update_producers_authority` to prevent chain from stopping.

### CODE REVIEW

**Review by swatanabe — CHANGES_REQUESTED** [review]

File: libraries/chain/controller.cpp
Comment (heifner) [reply libraries/chain/controller.cpp:2304]: Yes, this is a hard-forking change and I should have mentioned that in the description. Obviously no chain has removed these permissions or they would have hit this issue. Your point is, of course, that once this is merged those with this change will continue a chain with the permission removed and those without the change will not. I hate to add a protocol feature for something like this. I guess another option would be to subjectively not allow removal of `prod.major` and `prod.minor`. Any other ideas?

Comment (swatanabe) [reply libraries/chain/controller.cpp:2304]: I think that subjectively disallowing the removal is the correct behavior.  This is assuming that my conclusion that a block removing the permissions cannot currently become irreversible is correct.  I'd like a second opinion confirming that before committing to a course of action.
- It is impossible to build on such a block because `start_block` always fails
- A block can only become irreversible after a descendant is validated

Therefore, I think the following semantics will preserve consensus safely:
- Reject blocks containing the removal
- Reject normal transactions containing the removal
- Subjectively reject deferred transaction containing the removal (cannot retire with objective fail)

This matches the semantics of subjective transaction failure.

Comment (swatanabe) [reply libraries/chain/controller.cpp:2304]: Oh, wait...

Removing and then recreating the permission is a thing.

- When producing a block subjectively reject transactions containing the removal
- When validating a block, reject the block if the permissions do not exist when the block is completed

**Pull request closed without merging**
